gm4_crossbow_cartridges/data/gm4_crossbow_cartridges/functions/projectile/arrow_motion.mcfunction

```diff
@@ -1,6 +1,8 @@
 #run from check_projectile
 # @s = player using the crossbow
 
+tag @s add gm4_cb_use
+
 #copy motion of arrow to projectile and kill arrow
-data modify entity @e[tag=gm4_cb_proj,distance=..2,limit=1,sort=nearest] Motion set from entity @e[type=minecraft:arrow,distance=..2,limit=1,sort=nearest] Motion
-kill @e[type=minecraft:arrow,distance=..2,limit=1,sort=nearest]
+data modify entity @e[tag=gm4_cb_proj,distance=..2,sort=nearest,limit=1] Motion set from entity @e[tag=gm4_cb_arrow,distance=..2,sort=nearest,limit=1] Motion
```

Comment:
this should include a `type=arrow` check along with the tag. Otherwise it will search through every entity. This is true for all cases, so you can do a find replace `tag=gm4_cb_arrow` -> `type=arrow,tag=gm4_cb_arrow`